extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/MongoDBSinkBuilder.java

```diff
@@ -19,173 +19,170 @@
-
-        SupplierEx<MongoClient> localConnectionSupplier = connectionSupplier;
-        FunctionEx<MongoClient, MongoDatabase> localDatabaseFn = databaseFn;
-        FunctionEx<MongoDatabase, MongoCollection<T>> localCollectionFn = collectionFn;
-        ConsumerEx<MongoClient> localDestroyFn = destroyFn;
-
-        boolean localOrdered = ordered;
-        boolean localBypassValidation = bypassValidation;
-
-        return SinkBuilder
-                .sinkBuilder(name, ctx -> {
-                    MongoClient client = localConnectionSupplier.get();
-                    MongoCollection<T> collection = localCollectionFn.apply(localDatabaseFn.apply(client));
-                    return new MongoSinkContext<>(client, collection, localDestroyFn, localOrdered, localBypassValidation);
-                })
-                .<T>receiveFn(MongoSinkContext::addDocument)
-                .flushFn(MongoSinkContext::flush)
-                .destroyFn(MongoSinkContext::close)
-                .preferredLocalParallelism(preferredLocalParallelism)
-                .build();
-
-    }
-
-    private static class MongoSinkContext<T> {
-
-        final MongoClient client;
-        final MongoCollection<T> collection;
-        final ConsumerEx<MongoClient> destroyFn;
-        final InsertManyOptions insertManyOptions;
-
-        final List<T> documents;
-
-        MongoSinkContext(
-                MongoClient client,
-                MongoCollection<T> collection,
-                ConsumerEx<MongoClient> destroyFn,
-                boolean ordered,
-                boolean bypassValidation
-        ) {
-            this.client = client;
-            this.collection = collection;
-            this.destroyFn = destroyFn;
-            this.insertManyOptions = new InsertManyOptions()
-                    .ordered(ordered)
-                    .bypassDocumentValidation(bypassValidation);
-
-            documents = new ArrayList<>();
-        }
-
-        void addDocument(T document) {
-            documents.add(document);
-        }
-
-        void flush() {
-            collection.insertMany(documents, insertManyOptions);
-            documents.clear();
-        }
-
-        void close() {
-            destroyFn.accept(client);
+        checkNotNull(documentIdentityFn, "documentIdentityFn must be set");
+
+        final SupplierEx<MongoClient> connectionSupplier = this.connectionSupplier;
+        final Class<T> documentClass = this.documentClass;
+        final String databaseName = this.databaseName;
+        final String collectionName = this.collectionName;
+        final FunctionEx<T, String> selectDatabaseNameFn = this.selectDatabaseNameFn;
+        final FunctionEx<T, String> selectCollectionNameFn = this.selectCollectionNameFn;
+        final FunctionEx<T, Object> documentIdentityFn = this.documentIdentityFn;
+        final String fieldName = this.idFieldName;
+        final ConsumerEx<ReplaceOptions> updateOptionsChanger = this.replaceOptionsChanger == null
```

Comment:
otherwise whole builder will be serialized, as accessing fields inside lambda serializes also whole outer class. If you have local variable, only this variable will be serialized, without reference to outer class (because it's not a field)